exoscale/loadbalancer.go

```diff
@@ -85,30 +85,37 @@ func (l *loadBalancer) EnsureLoadBalancer(ctx context.Context, _ string, service
 	// (see https://github.com/kubernetes/kubernetes/issues/45234 for an explanation of the problem).
 	// The list of Nodes passed as argument to this method contains *ALL* the Nodes in the cluster, not only the
 	// ones that actually host the containers targeted by the Service.
-	// All Nodes are expected to be members of the same Instance Pool, so we only need to look the first one up.
 	if getAnnotation(service, annotationLoadBalancerServiceInstancePoolID, "") == "" {
+		debugf("no NLB service Instance Pool ID specified in Service annotations, inferring from cluster Nodes")
+
+		instancePoolID := ""
+		instancePools := make(map[string]struct{})
 		for _, node := range nodes {
 			instance, err := l.fetchComputeInstanceFromNode(ctx, node)
 			if err != nil {
 				return nil, err
 			}
 
 			if instance.Manager != "instancepool" {
-				return nil, fmt.Errorf("cluster node %q not running as an Instance Pool member", node.Name)
+				return nil, fmt.Errorf("cluster Node %q is not an Instance Pool member", node.Name)
```

Comment:
A cluster can contain instances that are not managed by instance pool (for statefull workload for instance).
This should not cause the LoadBalancer creation to fail.

Reply:
@7fELF should we simply skip those non-Instance Pool members Nodes?

Reply:
yes, but it should handle the case where there is no instance pool

Reply:
Suggestions implemented in 76a2da7.